internal/pkg/addon/package_test.go

```diff
@@ -0,0 +1,187 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package addon
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
+)
+
+type addonMocks struct {
+	// uploader *mocks.Mockuploader
+	// ws *mocks.MockworkspaceReader
+}
+
+func TestPackage(t *testing.T) {
+	const (
+		wlName = "mock-wl"
+		wsPath = "/"
+		bucket = "mockBucket"
+	)
+
+	tests := map[string]struct {
+		inTemplate  string
+		outTemplate string
+		setupMocks  func(m addonMocks)
+	}{
+		"AWS::Lambda::Function, zipped file": {
+			inTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::Lambda::Function
+    Properties:
+      Code: lambda/index.js
+      Handler: "index.handler"
+      Timeout: 900
+      MemorySize: 512
+      Role: !GetAtt "HelloWorldRole.Arn"
+      Runtime: nodejs12.x
+`,
+			outTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::Lambda::Function
+    Properties:
+      Code:
+        S3Bucket: mockBucket
+        S3Key: TODO
+      Handler: "index.handler"
+      Timeout: 900
+      MemorySize: 512
+      Role: !GetAtt "HelloWorldRole.Arn"
+      Runtime: nodejs12.x
+`,
+		},
+		"AWS::Glue::Job, non-zipped file": {
+			inTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::Glue::Job
+    Properties:
+      Command:
+        ScriptLocation: lambda/index.js
+`,
+			outTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::Glue::Job
+    Properties:
+      Command:
+        ScriptLocation: s3://mockBucket/TODO
+`,
+		},
+		"AWS::CodeCommit::Repository, directory without slash": {
+			inTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::CodeCommit::Repository
+    Properties:
+      Code:
+        S3: lambda
+`,
+			outTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::CodeCommit::Repository
+    Properties:
+      Code:
+        S3:
+          Bucket: mockBucket
+          Key: TODO
+`,
+		},
+		"AWS::ApiGateway::RestApi, directory with slash": {
+			inTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::ApiGateway::RestApi
+    Properties:
+      BodyS3Location: lambda/
+`,
+			outTemplate: `
+Resources:
+  Test:
+    Metadata:
+      "testKey": "testValue"
+    Type: AWS::ApiGateway::RestApi
+    Properties:
+      BodyS3Location:
+        Bucket: mockBucket
+        Key: TODO
+`,
+		},
+		"AWS::AppSync::Resolver, multiple replacements in one resource": {
```

Comment:
nice! I wonder if we should have a task to add `integ` tests that call [`cfn.ValidateTemplate`](https://docs.aws.amazon.com/AWSCloudFormation/latest/APIReference/Welcome.html) for each transformed resource type to ensure that our transformed keys are valid

Reply:
this would be cool, i'll add it as a todo item before we wrap this feature up